pandas/core/ops/__init__.py

```diff
@@ -457,7 +457,9 @@ def _should_reindex_frame_op(
         # TODO: any other cases we should handle here?
         cols = left.columns.intersection(right.columns)
 
-        if len(cols) and not (cols.equals(left.columns) and cols.equals(right.columns)):
+        if len(cols) and not (
+            cols.equals(left.columns.unique()) and cols.equals(right.columns.unique())
+        ):
```

Comment:
do we have a test that fails without this change?

Reply:
Yes, but the builds are gone. I will run the tests in the evening to find the relevant tests.

Reply:
https://github.com/pandas-dev/pandas/blob/898580169d51267ff5053c2c6ad69c9cd0ac184f/pandas/tests/frame/test_nonunique_indexes.py#L10

This one is crashing in line 255 because of memory issues. The condition is never fullfilled, so it blows up

Reply:
the condition in master is never fulfilled or the condition in the PR?

btw, usually let the person who asked the question hit the "resolve conversation" button

Reply:
Sorry, did not know that. Will do that in the future.

The input ``df`` has columns ``[A,A]``. When intersection is unique, then ``cols=[A]``, while ``left.columns=[A,A]`` and ``right.columns=[A,A]``. Without the change adding the unique, ``cols.equals(left.columns) and cols.equals(right.columns)`` will always be False. So we always return ``True``, which results in duplicating the columns for every passthrough, hence the memory explosion.

Reply:
is this still needed?

we *now* guaranteee that intersection returns unique columns, right so this should no longer be the case.

Reply:
Yeah, this is the problem. Is intersection is unique, cols.equals(left.columns) won't be True. This leads to a recursion in the test mentioned which blows up the memory, because we can not exit this.

Reply:
ok pls add some comments to this effect then.

Reply:
i would prob calculate left_uniques and right_uniques and make them variables as a bit simpler to read

Reply:
Thx, changed it and added a comment